packages/reports/src/Components/Routes/__tests__/route-with-sub-routes.spec.tsx

```diff
@@ -31,18 +53,78 @@ const route = {
 
 const MockRouteWithSubRoutes = () => <RouteWithSubRoutes {...route} />;
 
-describe('RouteWithSubRoutes component', () => {
+describe('<RouteWithSubRoutes />', () => {
     it('should render the "RouteWithSubRoutes" component', () => {
         render(<MockRouteWithSubRoutes />);
         const span_element = screen.getByText(/path param: \/test-path/i);
         expect(span_element).toBeInTheDocument();
     });
-
     it('should render properties', () => {
         render(<MockRouteWithSubRoutes />);
         const path_param = screen.getByText(/\/test-path/i);
         const exact_param = screen.getByText(/exact param: true/i);
         expect(path_param).toBeInTheDocument();
         expect(exact_param).toBeInTheDocument();
     });
+    it('should redirect to the specified path', () => {
+        render(<MockRouteWithSubRoutes />);
+        const redirect_element = screen.getByText(/Redirect to:/i);
+        expect(redirect_element).toBeInTheDocument();
+    });
+    it('should call redirectToLogin when not logged in and not logging in', () => {
+        const new_route = {
+            ...route,
+            is_logging_in: false,
+            is_logged_in: false,
+            is_authenticated: true,
+        };
+        render(<RouteWithSubRoutes {...new_route} />);
+        expect(screen.getByText(/Redirect to:/i)).toBeInTheDocument();
+    });
+    it('should set document title to default title when getTitle is not defined', () => {
+        render(<MockRouteWithSubRoutes />);
+        expect(document.title).toBe('| Default Title');
+    });
+    it('should set document title based on route.getTitle', () => {
+        const title = 'Test Title';
+        route.getTitle.mockReturnValue(title);
+        render(<MockRouteWithSubRoutes />);
+        expect(document.title).toBe(`Test Title | Default Title`);
+    });
+    it('should redirect to default subroute if pathname matches route path', () => {
+        const default_subroute = {
+            path: '/default-subroute',
+            default: true,
+            component: jest.fn(),
+            getTitle: jest.fn(),
+        };
+        const new_route = {
+            ...route,
+            routes: [default_subroute],
+            is_logging_in: false,
+            is_logged_in: true,
+            component: () => <div>Component</div>,
+        };
+
+        Object.defineProperty(window, 'location', {
+            writable: true,
+            value: { pathname: '/test-path' },
+        });
```

Comment:
```suggestion
        history.push('/test-path');
```
You will need `const history = createMemoryHistory();` declared above for this to work:)
Will this work for you?)